### DISCUSSION TITLE
Supported cloud service providers (status)

### DISCUSSION THREAD

**Discussion by seokho-son** [cloud-barista/cb-tumblebug#429] [Show and tell]
## CB-Tumblebug 연동 클라우드 현황 및 시험 정보 (v0.7.0)
- v0.7.0 릴리스 타겟: AWS, Azure, GCP, Alibaba, Tencent, IBM, Cloudit

## CB-Tumblebug FW 의존성 및 버전
- CB-Spider: https://github.com/cloud-barista/cb-spider/releases/tag/v0.7.0
- CB-Dragonfly: https://github.com/cloud-barista/cb-dragonfly/releases/tag/v0.7.0 (CB-TB 자동 제어 기능 필요시 활용)

## CB-Tumblebug 연동 현황

  | CloudOS (CSP) | 연동현황 | 시험현황 (생성 및 상태 제어, 모니터링) | 비고 |
  |:-------------:|:-------------:|:-------------:|:-------------:|
  | **AWS**            | 완료                             | 완료, 완료          | -                      | 
  | **Azure**          | 완료                             | 완료, 완료          | -                      | 
  | **GCP**            | 완료                             | 완료, 완료          | -                       | 
  | **Alibaba**       | 완료                             | 완료, 완료          | -                      | 
  | **Tencent**       | 완료 (#602)                            | 완료, 완료          | -                      | 
  | **IBM**             | 완료                             | 완료, 완료          | -                      | 
  | **OpenStack**        | 완료                               | 완료, 완료             | #431, #430, 시험 환경 한정 확인 | 
  | **Cloudit**        | 완료                               | 완료, 완료 (모니터링 오류 발견)             | 시험 상세: #529 (모니터링 오류 발견), 다중 VPC 생성에 이슈 있음 |
  | PoC **NCP(VPC)**    | 완료 (#439)                             | 완료 (https://github.com/cloud-barista/cb-tumblebug/discussions/1115#discussioncomment-3793430), 진행 중 (0.6.0기준 이슈가 발생했으나, 현황 확인은 필요)     | SG 룰 적용 이슈 있음 (NCP는 현재 SG 생성/삭제 API는 제공하지 않고 조회 API만 제공 https://github.com/cloud-barista/cb-tumblebug/discussions/439#discussioncomment-2438322), 외부 트래픽 처리 이슈 있음. 모니터링 이슈와 연계.        | 
  | PoC **NCP(Classic)**    | 완료                             | 완료 (https://github.com/cloud-barista/cb-tumblebug/discussions/439#discussioncomment-3793902), 미확인     |  -       | 
  | PoC **NHN Cloud**    | 완료 (#1038)                | 완료 (https://github.com/cloud-barista/cb-tumblebug/discussions/1062#discussioncomment-3793543), 미확인          |  -          | 
  | PoC **KT Cloud**    | 완료 (#953)                             | 완료, 미확인          | 리전없이 존으로 구분 TB의 다양한 스크립트 실행 가능 여부 확인 필요.           | 
  | **Mock**                 | 완료                             | 완료, 미적용          | 시험 이슈 #426               | 
  | **Cloud-Twin**      | 완료                             | 완료, 미적용           | 연동 시험용 환경 한정 |

- 연동시험(기본): 기본적인 MCIR 및 MCIS 라이프사이클 제어 기능의 동작 확인
- 연동시험(Agent 설치): MCIS 프로비저닝시, CB-Dragonfly의 모니터링 에이전트 자동 설치 확인

- 참고: [CB-Spider 자원 제공 현황](https://github.com/cloud-barista/cb-spider#3-%EC%A0%9C%EA%B3%B5-%EC%9E%90%EC%9B%90)

**Comment by powerkimhub** [comment]
@seokho-son @jihoon-seo (@hyokyungk)

#### CloudIt 시험 관련

- 다음 사항 먼저 검토 후 시험 추진해보시기 바랍니다. 
- 크리덴셜 정보는 별도로 전달하겠습니다.

- 다음 링크에서 VPC 특이사항 참고 
  - https://github.com/cloud-barista/cb-spider/blob/master/README.md#%EC%A0%9C%EA%B3%B5-%EC%9E%90%EC%9B%90
 - 세부 내용 필요시 다음 링크 참조
   - https://github.com/cloud-barista/cb-spider/issues/319

**Reply by seokho-son** [reply]
@powerkimhub 

💬 특이사항: 
    - VPC: 단일 VPC 생성 제공 (두개 이상 VPC 생성 요청시 동작을 보장할 수 없음)
    - Subnet: 단일 VPC에 Subnet 추가/삭제 가능
    - VPC 및 Subnet CIDR: 사용자의 설정값과 무관하게, CSP 내부에서 유휴 CIDR 할당 후 반납
    
이 해결될 가능성? 개선 계획이 있으신가요?
CB-TB에서는 대부분 동일한 클라우드라고 가정하고 처리하다 보니.. CloudIt에 한해서만 다른 방식으로 처리하는 것이 까다로운 측면이 있습니다. Subnet CIDR 도 클라우드 마다 다르게 자동 할당하는 형태도 활용하고 있어서, CB-TB에 바로 붙여보는 것에는 이슈가 있겠습니다.

단순히 단일 MCIS 까지 연동하는 것은 테스트할 수 있을 것 같은데, 
현재 붙여보는 것이 의미가 있을지는 잘 모르겠네요.

**Reply by powerkimhub** [reply]
@seokho-son 

- 현재, 1개의 VPC 이름과 여러개의 Subnet 이름은 사용자가 설정한 대로 생성해주고, PublicIP를 붙여줌
- Cloudit 내부에서만 사용되는 private IP 관련 CIDR만 설정한 CIDR와 다른 영역으로 제공 해주고 있는데,
- 현재 시험 시나리오에서 private ip를 직접 활용하지 않는다면, 소규모 시험은 가능하리라 생각됩니다. 
- 시험 시나리오 수정 없는 범위에서 시험 추진해서 결과를 확보하는 게 좋을 듯합니다. 
- 시험 결과에는  다음 제약 조건 링크를 남겨 두시면 될듯해요. 
  - https://github.com/cloud-barista/cb-spider/blob/master/README.md#%EC%A0%9C%EA%B3%B5-%EC%9E%90%EC%9B%90

**Comment by seokho-son** [comment]
https://github.com/cloud-barista/cb-tumblebug/discussions/529

**Comment by jihoon-seo** [comment]
[PoC 관련 클라우드 개발 및 시험을 위한 사전 준비 사항]
- 각 private repo에 접근할 수 있는 GitHub 계정 + private repo 접근용으로 생성한 Personal Access Token 을 사전에 준비해야 함. (private repo 접근 권한 보유 멤버 한정)

[PoC 상태인 CB-Spider 드라이버의 빌드 방법]
1. CB-Spider 소스코드 디렉토리에서 `make dyna` 실행 → CB-Spider 바이너리 및 PoC 외 드라이버가 빌드됨
2. `cd utils/driver-build-docker`
3. `./1.driver-build-container-run.sh` → CB-Spider PoC 드라이버 빌드용 컨테이너로 진입
4. `cd 2.build`
5. `./ncp-build.sh`, `./ncpvpc-build.sh`, `./nhncloud-build.sh` 실행 (GitHub 계정 및 Personal Access Token Password 입력 필요)
6.  CB-Spider 실행 (dynamic 모드로): cloud-barista/cb-spider/bin$ `./start-dyna.sh`
